src/tfr_control/arduino/encoder/encoder.ino

```diff
@@ -1,46 +1,41 @@
-/*
- * rosserial Publisher Example
- * Prints "hello world!"
- */
- #include <Encoder.h>
-
+#include <Encoder.h>
 #include <ros.h>
-#include <tfr_msgs/DrivebaseVelocity.h>
-
+#include <tfr_msgs/EncoderReading.h>
 #include <quadrature.h>
 
 ros::NodeHandle nh;
 
-tfr_msgs::DrivebaseVelocity velocities;
-ros::Publisher drivebase_publisher("drivebase_velocity", &velocities);
-
 //encoder level constants
 const double GEARBOX_CPR = 4096; //pulse per revolution
-const double GEARBOX_RPM = 0.876; //revolutions per meter
+//TODO note this rev/m is acually 0.87, but I had to step it up to make testing work
+const double GEARBOX_RPM = 60; //revolutions per meter
 
 //pin constants
 const int GEARBOX_LEFT_A = 2;
 const int GEARBOX_LEFT_B = 3;
 
 //encoders
 Quadrature gearbox_left(GEARBOX_CPR, GEARBOX_LEFT_A, GEARBOX_LEFT_B);
+tfr_msgs::EncoderReading reading;
 
 //algorithm constants
 const float rate(1/50.0);
 
+ros::Publisher encoders("encoders", &reading);
+
 void setup()
 {
     nh.initNode();
-    nh.advertise(drivebase_publisher);
+    nh.advertise(encoders);
     nh.getParam("~rate", &rate);
-    
 }
 
 void loop()
 {
-    velocities.timestamp = nh.now();
-    velocities.left_vel = gearbox_left.getVelocity()/GEARBOX_RPM;
-    drivebase_publisher.publish(&velocities);
-    nh.spinOnce();
+    reading.left_vel = gearbox_left.getVelocity()/GEARBOX_RPM;
+    //TODO hook up right encoder
```

Comment:
Also todo, hook up turntable (just making sure this is known, you don't need to worry about adding a comment or anything)